negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationAPIService.java

```diff
@@ -306,10 +319,22 @@ public ContractNegotiation handleContractNegotiationAgreed(String contractNegoti
 				credentialUtils.getConnectorCredentials());
 		if(response.isSuccess()) {
 			log.info("Updating status for negotiation {} to agreed", contractNegotiation.getId());
-			ContractNegotiation contractNegtiationAgreed = contractNegotiation.withNewContractNegotiationState(ContractNegotiationState.AGREED);
-			contractNegotiationRepository.save(contractNegtiationAgreed);
 			log.info("Saving agreement..." + agreementMessage.getAgreement().getId());
 			agreementRepository.save(agreementMessage.getAgreement());
+			
+			ContractNegotiation contractNegtiationAgreed = ContractNegotiation.Builder.newInstance()
```

Comment:
Can this one be changed with withNewContractNegotiationState(ContractNegotiationState.AGREED)?

Reply:
I just verified in the changed code, there is agreement added from AgreementMessage, not previous contract negotiation